piff/input.py

```diff
@@ -572,7 +613,7 @@ def readStarCatalog(self, cat_file_name, cat_hdu, x_col, y_col, flag_col, use_co
             cat = cat[cat[use_col]!=0]
 
         # Limit to nstars objects
-        if nstars is not None and nstars > len(cat):
+        if nstars is not None and nstars < len(cat):
             logger.info("Limiting to %d stars for %s",nstars,cat_file_name)
             cat = cat[:nstars]
```

Comment:
(This is something I missed the previous time around, sorry.) Do we really want it to just take the first nstars? I think the sextractor catalogs at least are built up in ascending x, y coordinates so that you would just get one edge of the CCD this way. We could just do np.random.choice(len(cat), nstars, replace=False) though I dunno if we like the randomness.